http-clients/src/main/java/com/palantir/remoting3/clients/UserAgent.java

```diff
@@ -55,69 +49,41 @@ static UserAgent of(String serviceName, String instanceId, String version) {
     static UserAgent of(String serviceName, String version) {
         Agent agent = ImmutableAgent.builder()
                 .serviceName(serviceName)
-                .version(version)
+                .version(UserAgents.isValidVersion(version) ? version : DEFAULT_VERSION)
                 .build();
         return ImmutableUserAgent.builder()
                 .addAgents(agent)
                 .build();
     }
 
-    static UserAgent append(UserAgent userAgent, String serviceName, String version) {
-        Agent agent = ImmutableAgent.builder()
-                .serviceName(serviceName)
-                .version(version)
-                .build();
+    /**
+     * Returns the {@link UserAgent} comprising all {@link UserAgent#agents} from the left and all {@link
+     * UserAgent#agents} from the right given {@link UserAgent}s.
+     */
+    default UserAgent merge(UserAgent other) {
```

Comment:
@lynchmattj @diogoholanda didn't like this the first time around. I'm still in favor of this version

Reply:
am fine with either, just want to keep the `base.addstuff(<stuff>)` syntax over `UserAgents.merge(base, other)`.

Reply:
well, it's incomplete because it was missing the `append(service, node, version)` variant.

Reply:
I don't think `append(service, node, version)` makes sense to append. when would a user agent have more than one `node`?

@markelliot the thing they explicitly didn't like was the non static version

Reply:
oh I see, because we want the agent construction to become more structured.

the non-static version makes sense to me, guess we can wait for one of them to chime in?

Reply:
i'd like to move on here since the qos work is blocking on it and there's still 3-4 PRs and releases that need to trickle through before I can pick up and make those changes.

Reply:
Regardless of the static method issue, I'm not sure you actually want to be able to merge two UserAgents. I don't know what having multiple nodeIds means in that case.

Reply:
ok. then the type should be 
```
class UserAgent {
  Optional<String> node;
  List<Agent> libraries;
}
```

Reply:
used as:
```
UserAgent agent = UserAgent.of("my-service", "192.168.0.1", "1.2.3")
  .addLibrary(MyApi.class);
  .addLibrary("my-other-lib", "4.5.5");
```
or similar

Reply:
pretty close to what we had, i guess

Reply:
sounds great to me. there was some push back on having UserAgent accept a `Class` and do the manifest business. may just want to have the `add(name, version)` for now

Reply:
deal

Reply:
sorry, the other option here would be:

    Agent primary;
    List<Agent> informational;

Reply:
...which also eliminates the dependency on the list ordering

Reply:
ok, try again.